### PR TITLE
SET fixes

### PR DESCRIPTION
**Pull request by Kelt**
fixes for SET

### CODE REVIEW

**Comment by Kelt** [comment]
@guyca @enterline

File: android/app/src/main/java/com/reactnativenavigation/utils/ViewUtils.java [hunk 1]
```diff
@@ -185,7 +185,7 @@ private static int getStatusBarHeight() {
     }
 
     public static ForegroundColorSpan[] getForegroundColorSpans(TextView view) {
-        SpannedString text = (SpannedString) view.getText();
+        SpannedString text = new SpannedString(view.getText());
```
Comment (guyca) [inline android/app/src/main/java/com/reactnativenavigation/utils/ViewUtils.java]: What's the motivation behind this change?

File: android/app/src/main/java/com/reactnativenavigation/views/sharedElementTransition/SharedElementTransition.java [hunk 2]
```diff
@@ -80,7 +80,7 @@ private void saveTextViewSpannedText(final TextView view) {
             @Override
             public void run() {
                 if (spannableText == null) {
-                    spannedText = (SpannedString) view.getText();
+                    spannedText = new SpannedString(view.getText());
```
Comment (guyca) [inline android/app/src/main/java/com/reactnativenavigation/views/sharedElementTransition/SharedElementTransition.java]: Why create a new SpannedString? Is there a performance hit from creating a new instance of SpannedString?

Comment (Kelt) [inline android/app/src/main/java/com/reactnativenavigation/views/sharedElementTransition/SharedElementTransition.java]: finally was able to repro ... so, if i wrap TextInput in SharedElementTransition, i get 

java.lang.ClassCastException: android.text.SpannableStringBuilder cannot be cast to android.text.SpannedString at com.reactnativenavigation.views.sharedElementTransition.SharedElementTransition$2.run(SharedElementTransition.java:80)

since view is com.facebook.react.views.textinput.ReactEditText and view.getText() returns SpannableStringBuilder

This however doesnt happen if I wrap TextInput in transparent View

Comment (guyca) [inline android/app/src/main/java/com/reactnativenavigation/views/sharedElementTransition/SharedElementTransition.java]: The exception isn't thrown when you wrap the Text in a transparent view since in that case We regard the shared element as a simple view - which doesn't have a text color property so this entire logic isn't executed.

I wonder under which condition `getText()` returns a builder and when does it return the concrete SpannedString. Can you confirm this has no performance hit?
